res/css/views/settings/tabs/user/_GeneralUserSettingsTab.pcss

```diff
@@ -37,12 +37,27 @@ limitations under the License.
 
 .mx_GeneralUserSettingsTab_accountSection .mx_EmailAddresses,
 .mx_GeneralUserSettingsTab_accountSection .mx_PhoneNumbers,
-.mx_GeneralUserSettingsTab_discovery .mx_ExistingEmailAddress,
-.mx_GeneralUserSettingsTab_discovery .mx_ExistingPhoneNumber,
+.mx_GeneralUserSettingsTab_discovery .mx_GeneralUserSettingsTab_discovery_existing,
 .mx_GeneralUserSettingsTab_languageInput {
     margin-inline-end: var(--SettingsTab_fullWidthField-margin-inline-end);
 }
 
+.mx_GeneralUserSettingsTab_discovery_existing {
+    display: flex;
+    align-items: center;
+    margin-bottom: 5px;
+
+    .mx_GeneralUserSettingsTab_discovery_existing_address,
+    .mx_GeneralUserSettingsTab_discovery_existing_promptText {
+        flex: 1;
+        margin-right: 10px;
+    }
+
+    .mx_GeneralUserSettingsTab_discovery_existing_button {
+        margin-left: 5px;
+    }
```

Comment:
Is there any reason to nest those ?
This artifically increases the specificity of the selector and forces a specific DOM structure.

Reply:
I do not really see the merit of not forcing a specific DOM structure here (because the style rules are supposed to be used on very this place), though.